### PR TITLE
mobile/ci: Further fix for docs publishing

### PR DESCRIPTION
**Pull request by phlax**
Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

File: .github/workflows/mobile-docs.yml [hunk 1]
```diff
@@ -30,20 +30,32 @@ jobs:
...
+        path: mobile-docs
+        ssh-key: ${{ secrets.ENVOY_MOBILE_WEBSITE_DEPLOY_KEY }}
+    - run: mv mobile-docs ../mobile-docs
+      if: github.ref == 'refs/heads/main'
+
     - name: Add safe directory
       run: git config --global --add safe.directory "$GITHUB_WORKSPACE"
+    # Not sure if this is necessary
+    - name: Add safe directory
+      if: github.ref == 'refs/heads/main'
```
Comment (phlax) [inline .github/workflows/mobile-docs.yml:49]: i think lets test all steps up to and excluding actually pushing

File: .github/workflows/mobile-docs.yml [hunk 2]
```diff
@@ -30,20 +30,32 @@ jobs:
     runs-on: ${{ needs.env.outputs.agent_ubuntu }}
     timeout-minutes: 20
     steps:
+    # Checkout the Envoy repo
     - uses: actions/checkout@v4
+    # Checkout the envoy-mobile/envoy-mobile.github.io repo
+    - uses: actions/checkout@v4
+      if: github.ref == 'refs/heads/main'
+      with:
+        repository: envoy-mobile/envoy-mobile.github.io
```
Comment (abeyad) [inline .github/workflows/mobile-docs.yml:39]: are the docs not moved to the envoy repo?  i thought we're completely done with the envoy-mobile repo?

Comment (phlax) [reply .github/workflows/mobile-docs.yml:39]: not at this point but thats defo what i think we should

it requires a little bit more thought/discussion tho and `main` is breaking on this so probs we need to address

i just dont want to iterate this more than we have to 8/

Comment (abeyad) [reply .github/workflows/mobile-docs.yml:39]: makes sense to get this working first, i just didn't realize docs hadn't been moved over to the envoy repo, so was puzzled by it.

**Review by abeyad — APPROVED** [review]

**Comment by phlax** [comment]
@abeyad gonna work on this a bit further to get a bit more assurance that it will work when landed

**Comment by phlax** [comment]
i think this should work now - it tests publishing in PRs up to the commit 

```console
[master 4353a9d11] docs envoy-mobile@325f95dbdd9f53d58ce30b338de8c457a2ff7386
 41 files changed, 444 insertions(+), 416 deletions(-)
 create mode 100644 docs/envoy-mobile/latest/_static/js/html5shiv-printshiv.min.js
 create mode 100644 docs/envoy-mobile/latest/_static/js/html5shiv.min.js
Not pushing for pull request
```

the only bit i cant test easily is the ssh key - this PR changes how it is loaded - but im reasonably confident that it should work

**Review by abeyad — APPROVED** [review]
thanks @phlax !

**Pull request merged**